internal/integration/postgres_test.go

```diff
@@ -421,6 +427,94 @@ schema "test" {
 	})
 }
 
+func TestPostgres_CLI(t *testing.T) {
+	// Required to have a clean "stderr" while running first time.
+	c := exec.Command("go", "run", "-mod=mod", "ariga.io/atlas/cmd/atlas")
+	require.NoError(t, c.Run())
+	t.Run("SchemaInspect", func(t *testing.T) {
+		pgRun(t, func(t *pgTest) {
+			t.dropTables("users")
+			h := `
+			schema "public" {
+			}
+			table "users" {
+				schema = "public"
+				column "id" {
+					type = "int"
+				}
+				primary_key {
+					columns = [table.users.column.id]
+				}
+			}`
+			var expected schema.Schema
+			err := postgres.UnmarshalSpec([]byte(h), schemahcl.Unmarshal, &expected)
+			require.NoError(t, err)
+			t.applyHcl(h)
+			cmd := exec.Command("go", "run", "-mod=mod", "ariga.io/atlas/cmd/atlas",
+				"schema",
+				"inspect",
+				"-d",
+				t.dsn(),
+			)
+			stdout, stderr := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
+			cmd.Stderr = stderr
+			cmd.Stdout = stdout
+			require.NoError(t, cmd.Run(), stderr.String())
+			var actual schema.Schema
+			err = postgres.UnmarshalSpec(stdout.Bytes(), schemahcl.Unmarshal, &actual)
+			require.NoError(t, err)
+			require.Empty(t, stderr.String())
+			require.Equal(t, expected, actual)
+		})
+	})
+	t.Run("SchemaApply", func(t *testing.T) {
+		pgRun(t, func(t *pgTest) {
+			t.dropTables("users")
+			h := `
+			schema "public" {
+			}
+			table "users" {
+				schema = "test"
+				column "id" {
+					type = "int"
+				}
+				primary_key {
+					columns = [table.users.column.id]
+				}
+			}`
+			f := "atlas.hcl"
+			err := ioutil.WriteFile(f, []byte(h), 0644)
+			require.NoError(t, err)
+			defer os.Remove(f)
+			cmd := exec.Command("go", "run", "-mod=mod", "ariga.io/atlas/cmd/atlas",
+				"schema",
+				"apply",
+				"-d",
+				t.dsn(),
+				"-f",
```

Comment:
feels like these tests are similar enough to extract to a helper fn now?

Reply:
i would keep the assertions part of the test body, and not couple them to the helper though.

Reply:
You are right!, I merged